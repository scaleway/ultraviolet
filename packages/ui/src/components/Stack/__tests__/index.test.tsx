import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { Stack } from '..'

describe('Stack', () => {
  test(`should render correctly with default props`, () =>
    shouldMatchEmotionSnapshot(
      <Stack>
        <div>first child</div>
        <div>second child</div>
        <div>third child</div>
      </Stack>,
    ))

  test(`should render correctly with row direction`, () =>
    shouldMatchEmotionSnapshot(
      <Stack direction="row">
        <div>first child</div>
        <div>second child</div>
        <div>third child</div>
      </Stack>,
    ))

  test(`should render correctly with alignCenter`, () =>
    shouldMatchEmotionSnapshot(
      <Stack direction="row" alignItems="center">
        <div>first child</div>
        <div>second child</div>
      </Stack>,
    ))

  test(`should render correctly with wrap as boolean`, () =>
    shouldMatchEmotionSnapshot(
      <Stack direction="row" wrap>
        <div>first child</div>
        <div>second child</div>
      </Stack>,
    ))

  test(`should render correctly with wrap as string`, () =>
    shouldMatchEmotionSnapshot(
      <Stack direction="row" wrap="wrap-reverse">
        <div>first child</div>
        <div>second child</div>
      </Stack>,
    ))

  test(`should render correctly with width 100%`, () =>
    shouldMatchEmotionSnapshot(
      <Stack width="100%">
        <div>first child</div>
        <div>second child</div>
      </Stack>,
    ))

  test(`should render correctly max width 100%`, () =>
    shouldMatchEmotionSnapshot(
      <Stack maxWidth="100%">
        <div>first child</div>
        <div>second child</div>
      </Stack>,
    ))

  test(`should render correctly min width 100%`, () =>
    shouldMatchEmotionSnapshot(
      <Stack minWidth="100%">
        <div>first child</div>
        <div>second child</div>
      </Stack>,
    ))

  test(`should render different direction and gap on different viewport sizes`, () => {
    shouldMatchEmotionSnapshot(
      <Stack
        direction={{
          xxsmall: 'column',
          xsmall: 'column',
          small: 'row',
          medium: 'row',
          large: 'row',
          xlarge: 'row',
        }}
        gap={{
          xxsmall: 1,
          xsmall: 2,
          small: 3,
          medium: 4,
          large: 5,
          xlarge: 6,
        }}
      >
        <div>first child</div>
        <div>second child</div>
        <div>third child</div>
      </Stack>,
    )
  })
})
