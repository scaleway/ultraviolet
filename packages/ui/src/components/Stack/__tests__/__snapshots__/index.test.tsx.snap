// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stack > should render correctly max width 100% 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly min width 100% 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  min-width: 100%;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly with alignCenter 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
      <div>
        third child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly with row direction 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
      <div>
        third child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly with width 100% 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly with wrap as boolean 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render correctly with wrap as string 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap-reverse;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`stack > should render different direction and gap on different viewport sizes 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
}

@media (min-width: 1440px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width: 1280px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width: 1080px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width: 1800px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width: 720px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width: 320px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width: 1440px) {
  .emotion-0 {
    gap: 2.5rem;
  }
}

@media (min-width: 1280px) {
  .emotion-0 {
    gap: 2rem;
  }
}

@media (min-width: 1080px) {
  .emotion-0 {
    gap: 1.5rem;
  }
}

@media (min-width: 1800px) {
  .emotion-0 {
    gap: 3rem;
  }
}

@media (min-width: 720px) {
  .emotion-0 {
    gap: 1rem;
  }
}

@media (min-width: 320px) {
  .emotion-0 {
    gap: 0.5rem;
  }
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <div>
        first child
      </div>
      <div>
        second child
      </div>
      <div>
        third child
      </div>
    </div>
  </div>
</DocumentFragment>
`;
